고객관리프로그램 마크업 1차 완료

designerbohyun opened the pull request:
## 전체 사용 설명
1. index.html 클래스명 위에 각 영역마다의 설명을 주석으로 적어두었습니다.
: css에서 찾아볼 경우 주석으로 동일하게 입력해두었으니, ctrl+f로 찾으시면 됩니다.

## 클래스 단위 설명
1. wrap-box : 모달창 제외 한 나머지 요소들을 감싸고 있는 박스
2. customer-container : 고객 명단 영역 
3. sc-wrap-box : 검색창과 고객 관리 내용 영역을 감싸고 있는 박스
4. customer-add-modal-wrap : 고객 명단 추가 버튼을 클릭할 경우, 보여지는 고객 명단 추가 모달창
5. customer-edit-modal-wrap : 고객 정보 수정버튼 클릭시, 보여지는 고객 정보 수정 모달창
6. customer-information-edit-modal-wrap : 고객 정보 수정을 하던 도중 취소를 클릭한 경우, 보여지는 모달창
7. customer-delete-modal-wrap : 고객 정보 수정하는 도중 삭제 클릭한 경우, 보여지는 모달창
8. customer-information-cancel-modal-wrap : 고객 추가를 하다 취소를 한 경우, 보여지는 모달창
9. customer-detail-modal-wrap : 고객 상세 정보 창
10. manage-add-modal-wrap : 고객 관리항목 추가 클릭시 , 보여지는 모달창
11. manage-edit-modal-wrap : 고객 관리항목 수정 클릭시, 보여지는 모달창 
12. manage-show-modal-wrap : 관리항목 상세정보 모달창
13. content-edit-cancel-modal-wrap : 관리항목 수정을 하던 도중 취소를 클릭한 경우, 보여지는 모달창
14. csutomer-delete-modal-wrap : 관리항목 삭제 클릭시, 보여지는 모달창
15. content-cancel-modal-wrap : 관리항목 추가하던 도중, 최소할 경우, 보여지는 모달창
16. content-edit-nocomplete-modal-wrap : 관리항목 수정을 완료하지 않은 경우, 모달창
17. content-add-nocomplete-modal-wrap : 관리항목 등록을 완료하지 않은 경우 모달창
18. content-delete-nocomplete-modal-wrap : 관리항목 삭제가 완료되지 않은 경우 모달창
19. content-delete-complete-modal-wrap : 관리항목 삭제가 완료된 경우 안내 모달창

## 클래스 단위 수정사항 - 2023년 02월 07일 기준
- 7,14번은 delete-before-modal-wrap으로 수정 / visible은 customer-delete-before(7번), content-delete-before(14번)으로 구분
- 8,15번은 input-cancel-modal-wrap으로 수정 / visible은 customer-input-cancel(8번), input-cancel(15번)으로 구분
- 16,17,18번은 content-nocomplete-modal-wrap으로 수정
- 19번은 content-complete-modal-warp으로 수정

- visible 관련 코드는 각 해당되는 wrap 관련 스타일 하단에 작성 완료.

designerbohyun committed:
고객관리프로그램 마크업 1차 완료

Lai-khan commented:
고생했으~~ 👏👏👏

designerbohyun committed:
markup1 오타수정

designerbohyun committed:
markup1 디자인 오류 수정

 고객 관리 항목 추가시, 보여지는 모달창에서 날짜 영역 디자인 오류 수정사항입니다.

Lai-khan commented:
일정 크기 이하로 줄어들면 스타일이 깨지지 않도록 최소 크기 제한이 있어야 하는데 현재 마크업은 크기를 줄이면 스타일이 전부 깨집니다.

![image](https://user-images.githubusercontent.com/40736969/188463319-68592cd7-bc19-48a1-919f-0a0215c64923.png)

(와 요렇게도 된당)
![image](https://user-images.githubusercontent.com/40736969/188463908-9b79a990-5d33-4d22-b70d-28b8f6b397f4.png)


일단 개발쪽에서도 크기 축소 제한을 걸긴 하겠지만 일정 크기 이하로 줄어들어도 최소한의 스타일이 깨지지 않게 해주세요.
![image](https://user-images.githubusercontent.com/40736969/188463687-e9471d5b-c9fa-41cd-8ad9-cc77d9686b42.png)

요렇게 (가로 스크롤)

Lai-khan commented on css/common.css:
```diff
@@ -0,0 +1,36 @@
+    -ms-overflow-style: none;
+    }
+    
+   ::-webkit-scrollbar {
+     display: none;
```
스크롤바 숨겨달라고 했었나?
스크롤바 뜨는 부분이 왼쪽의 고객명단 밖에 없는데 마우스 대고 움직일때는 보이는 게 좋을 것 같아.

designerbohyun replied:
html, body에서 스크롤바 숨기는 기능을 없애면 되는데! 어디서 마우스를 대고 움직이는게 보이면 좋은거야??

Lai-khan replied:
고객 목록에 마우스 대고 휠 움직일 때!

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1252 @@
+                <div class="customer-information-cancel-modal">
+                    <div class="customer-information-cancel-modal-title">
+                        <div class="customer-information-cancel-warn-icon">
+                            <img src="./img/cancel_warn_icon.png" alt="">
```
여기랑 `관리항목 수정 취소 클릭 시`랑  `관리항목 취소 클릭 시` 부분이, 페이지 하단에 이렇게 뜨는데 뭔가 이상 생긴것 같음.
![image](https://user-images.githubusercontent.com/40736969/188467617-4582161f-d5d3-4401-b704-0858df097010.png)

아, 그리고 파일명 잘못됨. 저번에 calcle -> calcel로 고쳤는데 파일명은 안 고쳐서 깨지고 있음

Lai-khan replied:
근데 파일명을 수정해도 아이콘이 엄청 크게 나와서 결국 깨지더라.

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1252 @@
+                            <input id="manage-etc" type="text">
+                        </div>
+
+                    <div class="manage-button">
+                        <button class="manage-cancel">취소</button>
+                        <button class="manage-add">등록</button>
+                    </div>
```
그리고 모든 모달창의 취소 버튼이 이렇게 깨지고 있는데
![image](https://user-images.githubusercontent.com/40736969/188467926-8c6bc8cc-87cc-4565-95aa-5ba275d66fd1.png)

내가 보기엔 user agent stylesheet 때문인 것 같음.

각 브라우저마다 정해놓은 기본 규칙인데, 이 user agent stylesheet에서 명시하는 속성을 직접 override 하지 않으면 기본 규칙이 강제로 적용되는 거임.

혹시 맥에서 작업해서 사파리 브라우저에서만 확인한 거 아님??

마크업 작업 할 때는 대표적인 브라우저(엣지, 사파리, 크롬, 파이어폭스 (가끔은 웨일까지도))에서 잘 돌아가는지 확인해주는게 좋아.

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1252 @@
+                
+                                <div class="delete-agree">
+                                    <input id="agree" type="checkbox">
+                                    <label for="agree" class="delete-agree-text">위 내용을 확인하셨으며 삭제에 동의합니다.</label>
```
아, 여기도 깨지더라.
![image](https://user-images.githubusercontent.com/40736969/188468721-4281b592-59e9-490b-9d33-7af0f2fe4730.png)

Lai-khan reviewed (commented):
마크업 적용하려고 하는데 깨지는 부분이 좀 있어서 코멘트 남겼으
빠르게 수정 부탁해용~ 😘

designerbohyun commented:
웅웅! 확인해써! 최대한 빨리 수정하고 다시 공유할게!!!!!

designerbohyun committed:
고객관리프로그램 오류 수정 2차

1. 일정크기 이하로 줄어들 경우, 너비값,세로값 고정 : wrap-box의 너비,세로값을 주 사용 사이즈인 1920*1080px로 고정해뒀어(수정 완료)

2.스크롤바 숨기기
: 고객 목록에서 휠 움직일 경우 스크롤바 보이도록 수정 완료 및 커스텀 수정 완료

3. 이미지 및 취소,완료 버튼들 꺠지는 부분
: 지난번 클래스명 수정된게 일부는 수정이 안되어서 그랬던거 같아 지금은 내가 확인해서 수정 완료했는데! 혹시나 내가 또 놓친부분이 있을 수 있어서 언니가 작업하다가 오류나면 알려줘~!

designerbohyun commented:
언니가 위에 말해준 내용들은 수정 완료해서 다시 올렸어! 확인해주면 될거같아!!!ㅎㅎ
그리구 내가 이번주랑 다음주 주말에는 다 일정이 있어서ㅠㅠ 아무래도 회의는 어려울거 같은데.. 괜찮아?

Lai-khan commented:
> 언니가 위에 말해준 내용들은 수정 완료해서 다시 올렸어! 확인해주면 될거같아!!!ㅎㅎ 그리구 내가 이번주랑 다음주 주말에는 다 일정이 있어서ㅠㅠ 아무래도 회의는 어려울거 같은데.. 괜찮아?

그래. 나도 다음주까지 일정이 있어서 힘들것 같애
고객관리프로그램은 여기 PR에서 얘기하면 될것같고 냉장고는 다다음주에 와이어프레임 관련해서 논의하는걸로 하자

확인해보고 수정 요청사항 있으면 다시 코멘트 달게~

Lai-khan commented on css/common.css:
```diff
@@ -16,21 +16,26 @@ a{
 ol,ul,li{
     list-style: none;
-}
+} s
```
요건 오타인감?

designerbohyun replied:
요것도 수정 완료!

Lai-khan commented on css/style.css:
```diff
@@ -669,7 +670,7 @@
 }
 
-.delet-button-area-cancel{
+.delet-button-area-cancle{
```
아니.,.. 이건 왜.. 오타를 수정했으면서 다시 오타로 바꾼거여;;;
파일명을 바꿔야지...

designerbohyun replied:
다시..파일명이랑 오타수정해서 바꿨어~!

Lai-khan reviewed (commented):
일단 눈에 보이는 것만 적었으!
나머지는 적용시켜보고!

designerbohyun committed:
220926_오타수정및오류수정

1. common.css - s : 오타 삭제
2. cancle => cancel로 파일 수정 및 오타,오류 수정

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1252 @@
+
+                </ul>
+
+                <div class="customer-detail-menu">
+                    <div class="customer-detail-info">
+                        <p>고객 상세 정보</p>
+                    </div>
+
+                    <div class="customer-info-edit">
+                        <p>고객 정보 수정</p>
+                    </div>
+
+                    <div class="customer-delete">
+                        <p>고객 삭제</p>
+                    </div>
+                </div>
```
요 부분이 각 고객 리스트 별로 위치 조절 가능한 부분이야?
아니면 그냥 첫번째 고객 옆에 위치 고정이야?
각 고객 리스트 별로 그 옆에 뜨는 줄 알았는데 '고객 리스트 영역' 바깥쪽에 위치해 있어서

designerbohyun replied:
각 고객 리스트 별로 위치 조절이 가능한 부분인데, 고객 리스트 영역 내에 있으면 구조가 꼬여서 밖에 위치해두고, 따로 position으로 우선 첫번째 고객에게 뜨게 잡아뒀는데, 각 고객 리스트 별로 옆에 뜨게 하는건,...어떻게 해야할지 모르겠어서 우선 첫 고객에게만 둔거였어

Lai-khan replied:
오케. 그럼 내가 알아서 클릭할 때마다 위치 조정해서 옆에다가 붙여볼게.
고객 리스트 간의 간격(height)이 어느정도인지, position 조정 어케하면 되는지 가이드 가능?

designerbohyun replied:
.customer-detail-menu{    
    position: absolute;
    left: 98%;
}

내가 지금 이렇게 수정할건데, 간격을 정확하게 몇정도 떨군걸로 하는게 아니라서, 간격의 기준이 하단 이미지처럼 스크롤바가 딱 가려지도록 시점을 정해두고 position : absolute;로 정렬한거라서 위에 left: 98%로 잡아뒀구, position의 기준은 
.customer-container-customer-list-area(고객리스트 클래스명)이야.


![image](https://user-images.githubusercontent.com/104741286/195980553-bb4292f5-a4a4-4da5-ac91-ce65f32d610c.png)


혹시 이상하거나 이해 안가면 다시 comment 달아줭..!

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1252 @@
+
+
+            <!-- 고객 명단 영역 : 고객 리스트 정렬버튼 영역 -->
+            <div class="customer-container-list-align-area">
+                <div class="customer-list-align">
+                    <p>가나다순</p>
+                    <div class="img">
+                        <img src="./img/arrow-down.svg" alt="">
+                    </div>
+                </div>
```
요거는 왜 select-option 태그로 안하고 div로 한거야?
밑에 관리항목 검색도 그렇던데.

Lai-khan replied:
```html
<select>
	<option value="alphabetical">가나다순</option>
	<option value="recent">최신순</option>
</select>
```
혹시 이런식으로 변경 가능할까?

designerbohyun replied:
아~ 내가 저걸 몰라서 div로 했었어 수정해서 올릴게

designerbohyun replied:
언니 내가 입력을 해봤는데 문제가 select, option 태그를 사용하면 select 태그는 스타일 수정이 되는데, option 태그에서 사용한 경우는 디자인 수정이 안되는거 같더라구 그래도 우선 수정해서 올릴게~! 한번 보구 알려줘! 어떻게 하는게 좋을지

Lai-khan replied:
으잉?? option 태그라고 해서 다를게 없는데 스타일 수정이 안된다고??
아, option 태그는 적용할 수 있는 style이 몇개 안되는구나.

근데 이거 카톡 장애나서 불편하네. 티스토리가 안들어가져서 관련 자료 검색이 안된다..

Lai-khan replied:
https://www.w3schools.com/howto/howto_custom_select.asp
이거 참고해봐바. 커스텀 가능할 것 같은데

`select box custom`이라고 구글 검색해봐바

designerbohyun replied:
아~ 알겠어! 고마워!! 한번 확인해볼겡!

Lai-khan replied:
이거랑 밑에 관리항목 검색도 수정 부탁해~

designerbohyun committed:
221015_고객리스트 상세설정 영역, 고객 리슽 순서 정렬 영역 스타일 수정

.customer-detail-menu 영역 : 위치 조정 및 사이즈 조정

.customer-container-list-align-area 영역 : div -> select, option 태그로 수정 및 스타일 수정 작업

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1241 @@
+                    </div>
+    
+                    <div class="search-detail-list">
+                            <li class="saerch-detail-list1">
+                                <p>내용</p>
+                            </li>
+                            <li class="saerch-detail-list2">
```
여기 오타났네.
`saerch` -> `search`

Lai-khan replied:
여기 오타났네.
`saerch` -> `search`

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1241 @@
+                        <img src="./img/search-bar-clear-button.png" alt="">
+                    </div>
+                    <div class="search-bar-search-button">
+                        <img src="./img/serach-bar-search-button.png" alt="">
```
요기랑

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1241 @@
+                    </div>
+
+                    <div class="search-bar-search-button">
+                        <img src="./img/serach-bar-search-button.png" alt="">
```
요기두

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1241 @@
+                            <p>취소</p>
+                        </button>
+    
+                        <button class="edit-button-area-delet">
```
요기 오타 7개 
`delet` -> `delete`

Lai-khan reviewed (commented):
전체적으로 오타가 좀 있더라.
이것 말고도 더 있으니까 한번 더 점검해봐바~!

Lai-khan commented:
수정 좀....

designerbohyun committed:
221023_오타수정

delet -> delete로 오타 수정
deletee -> delete로 오타 수정
saerch -> search로 오타 수정

designerbohyun commented:
언니 우선 오타 수정은 다 했구, 늦게 확인해서 미안해..ㅠ select/option 스타일은 javascript를 사용해서 class명을 줘서 css로 스타일을 수정하는거 같아서 javascript를 더 확인해 봐야할거같아.

designerbohyun committed:
221109_반응형 작업 수정

가로 최소 사이즈 : 1200px
세로 최소 사이즈 : 800px

최대 큰 사이즈 : 사이즈 늘릴수록 풀화면
중간 사이즈 : 1650px ~ 1201px
최소 사이즈 : 1200px

- 1200px * 800px 이하로는 화면 사이즈 줄어들지 않음.

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1245 @@
+
+
+            <!-- 고객 명단 영역 : 고객 리스트 정렬버튼 영역 -->
+            <label class="customer-container-list-align-area">
+                <select class="customer-list-align">
+                    <option value="alphabetical">가나다순</option>
+                    <option value="recent">최신순</option>
+                </select>
+            </label>
```
```suggestion
<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="utf-8">
    <title></title>
    <style>
      .custom-select {
        position: relative;
        width:150px;
      }
      .custom-select select {
        display: none;
      }
      .selected {
        background-color: #E7EDFF;
      }
      .selected p {
        margin: 0;
      }
      .selected .img{
        width: 12px;
      }
      .select-items div,.selected {
        padding: 8px 16px;
        border: 1px solid transparent;
        border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent;
        display: flex;
        justify-content: space-evenly;
        align-items: center;
        cursor: pointer;
        user-select: none;
      }
      .select-items {
        position: absolute;
        background-color: #E7EDFF;
        top: 100%;
        left: 0;
        right: 0;
        z-index: 99;
      }
      .select-hide {
        display: none;
      }
    </style>
  </head>
  <body>
    <div class="custom-select">
      <select>
        <option value="alphabetical">가나다순</option>
        <option value="recent">최신순</option>
      </select>
      <div class="selected">
        <p>가나다순</p>
        <div class="img">
          <img src="./img/arrow-down.svg" alt="">
        </div>
      </div>
      <div class="select-items select-hide">
        <div class="item">가나다순</div>
        <div class="item">최신순</div>
      </div>
    </div>
    <script>
      let selected = document.getElementsByClassName("selected")[0];
      let items = document.getElementsByClassName("select-items")[0];
      selected.addEventListener("click", function(e) {
          e.stopPropagation();
          items.classList.toggle("select-hide");
      });
    </script>
  </body>
</html>
```

대충 이런식으로 간단하게 짜봤는데, 자바스크립트는 어차피 React에 적용하면서 달라질거라 클릭할때 펼쳐지는 부분만 대충 짰고, html이랑 css를 집중적으로 봐주면 될것 같아. css를 CRM꺼를 적용해보려고 했는데 음..힘들더라고;; 암튼 저런식으로 해서 선택박스 만들면 될것 같아!

```
<select></select> // hide
<div class="1"></div> // <select>처럼 보이는 부분
<div class="2"></div> // 위의 div 클릭시 option처럼 펼쳐지게 보이는 부분
```

여기랑 관리항목의 검색창 부분 2군데 수정 부탁해~!

designerbohyun replied:
아~!!! 알겠어!

designerbohyun committed:
221211_select,option&반응형수정

공유해준 select,option은 고객 정렬이랑 검색 필터구간에 적용하고, 반응형까지 우선 수정함. -> 다만, 반응형은 언니의 노트북에서도 한번 더 체크를 하는게 좋을 거 같음.

Lai-khan commented on css/style.css:
```diff
@@ -156,14 +155,159 @@
     z-index: 99;
+} */
+
+/* 고객 명단 영역 : 고객 리스트 정렬버튼 영역 수정본 */
```
어차피 예전 코드들은 커밋에 다 남으니까 굳이 주석처리하지 않고 삭제해도 될 것 같아~

designerbohyun replied:
웅~ 알겠어!

Lai-khan commented on index.html:
```diff
@@ -36,15 +36,24 @@ <h3 class="customer-container-title">고객명단</h3>
+                  <option value="recent">최신순</option>
                 </select>
-            </label>
+                <div class="selected">
+                  <p>가나다순</p>
+                  <div class="arrow">
+                      <svg><path d="M12 15.375 6 9.375 7.4 7.975 12 12.575 16.6 7.975 18 9.375Z"/></svg>
+                  </div>
+                </div>
+                <div class="select-items select-hide">
+                  <div class="item">가나다순</div>
+                  <div class="item">최신순</div>
+                </div>
```
![image](https://user-images.githubusercontent.com/40736969/207597292-d34dde76-94d8-44fd-bbcb-70397c0df1d8.png)
뭔가 좀... 삐뚠것 같애...
원래 저렇게 살짝 떨어지는 디자인이 의도한게 맞아?

designerbohyun replied:
흠...다시 확인해볼게 혹시 화면 사이즈 상관없이 사이즈 줄였다 늘려도 계속 틀어져서 보여?

Lai-khan replied:
응. 그리고 저 두 부분이 좀 붙어있었으면 좋겠엉

designerbohyun replied:
언니 우선 위치 조정은 수정했는데, 아무래도 저 부분 붙이는게 문제가 좀 있어서 시각적으로 붙이면 좀 이상해서 그냥 띄워두는게 훨씬 나을거 같아

designerbohyun committed:
221223_select.option 위치 조정

select, option 틀어져 보이는 부분은 수정 완료했지만, 시각적으로 붙이는 문제는 이상해보여서 오히려 떨어트리는게 낫다고 생각함. + 아예 시각적으로 상단 select에서 내려오는 시각으로 수정하려고 했으나 order이 먹지않는 문제가 발생 -> 이 부분은 스타일 변경을 아예 다시 해야하는 문제가 생겨서 떨어트리는 걸로 수정..:(

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+}
+
+/* customer 클릭했을때, hover-blue 영역이 보이게 유지 */
+.hover-blue.acitve{
```
오타 발견~
나중에 수정해줘

designerbohyun replied:
웅~!!! 내가 지금 밖이라서 집 가서 수정할겡

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+    margin-right: 27px;
+}
+
+.customer:hover .people-icon path{
+    fill: #2A61FF;
+}
```
고객 클릭 했을때
![image](https://user-images.githubusercontent.com/40736969/213845457-5e821684-4ef7-4ec3-bcf7-f8803ba38aee.png)
이렇게 하고싶은데 그러면 CSS에 요렇게 추가하면 되려나?
```
.customer .people-icon.active path {
	fill: #2a61ff;
}
```

designerbohyun replied:
내가 집 가서 확인해보궁!! 답변줄겡!!

designerbohyun replied:
웅 언니 css 언니가 작성해준대로 추가하면 돼!

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+    display: flex;
+    justify-content: left;
+    align-items: center;
+    cursor: default;
```
이거 pointer로 바꿀게~~

designerbohyun committed:
220121_active 오타 수정

acitve -> active 라고 오타 수정

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1261 @@
+                        </div>
+                        <div class="customer-list-name"><p>
+                            배 수한무 거북이와 두루미 삼천갑자 동</p></div>
+                        <div class="detail-icon">
+                            <img src="./img/detail-icon.svg" alt="">
+                        </div>
```
여기 영역이 너무 작아서 클릭 이벤트가 잘 안먹혀ㅠㅠ
클릭 영역을 좀만 여유있게 넓혀줘~
대충 그 부근을 눌러도 클릭했다고 인식할 수 있게!

Lai-khan replied:
![image](https://user-images.githubusercontent.com/40736969/213923371-8b32ada1-856c-4c6e-8d37-304e50b22373.png)
주황색 부분은 마진이고 클릭이벤트가 발생하는 부분은 저기 파란색 부분!
저 부분이 상하좌우로 조금씩 늘어나야 할 것 같아

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1261 @@
+                        <input class="number-input" type="text">
+                    </div>
+    
+                    <div class="customer-address">
+                        <span>주소</span>
+                        <input class="address-input" type="text">
+                    </div>
```
주소가 최대 50자인데 50자 입력하면 잘림.
![image](https://user-images.githubusercontent.com/40736969/213923146-af118e48-06ee-45db-a054-6831feab34f1.png)
글자 수에 따라 보여지는 칸이 늘어나야 할 것 같은데.

이 외에도 이름은 최대 20자, 관리항목 내용이나 메모는 200자 등 최대 글자수가 있으니까 한번씩 점검 부탁해!

designerbohyun replied:
언니 내가 이거 따로 지금 확인해보고 있는데, textarea로 수정하는거랑 scroll바 숨기는건 했는데, 언니거 원하는 글자수에 따라 보여지는 칸이 늘어나는거는 java-script로 진행하는거라서 그거는 내가 좀 더 알아보고 css로 할 수 있는 방법이 있으면 그걸로 하고, 아니면 js로 해볼게

Lai-khan reviewed (commented):
현재 개발, 고객 목록 조회, 고객 검색, 정렬, 고객 상세정보 조회까지 완료한 상태고
아래는 현재까지 개발하면서 발견한 개선 필요한 부분들이야!

고객 정보 추가, 수정, 삭제 개발하고 있을테니까 수정해주면 반영할게~

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+
+
+
+/* 고객 정보 수정 버튼 클릭시, 고객 정보 수정 모달창 */
+.customer-edit-modal-wrap{
+    display: none;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    min-width: 1200px;
+    min-height: 800px;
+    z-index: 999;
+    background-color: rgba(0, 0, 0, 0.3);
+}
+
+.customer-edit-modal-wrap.active{
+    display: block;
+}
+
+
+.customer-edit-modal{
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    width: 500px;
+    height: 500px;
+    border-radius: 20px;
+    box-sizing: border-box;
+    background-color: #fff;
+    box-shadow: 0 0 30px rgba(0, 0, 0, 0.1);
+    z-index: 999;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-evenly;
+    align-items: center;
+}
```
이 부분이 customer-add-modal 쪽 css랑 이름만 빼고 다 똑같아서 그냥 하나로 합칠게~
신규 생성이랑 수정이랑 형식이 거의 똑같은것 같아서 하나로 합쳐서 재사용하려고!

`customer-input-modal~~~` 뭐 이런식으로

Lai-khan replied:
요거 css는 하나로 합쳐서 사용하는데
개발상 문제가 살짝 있어서 실제로 사용하는 모달은 생성, 수정 분리하기로 했어.
🔔 noti~

아...잠만... 뭐가 또 문제가 생기는 것 같은데...흠..ㅠㅠ

designerbohyun replied:
무슨 문제가 생겼어?

Lai-khan replied:
개발쪽 문제야. 마크업은 상관 없는데..
따로 쓰자니 겹치는 부분이 너무 많고, 같이 쓰자니 상세정보, 생성, 수정의 데이터가 자꾸 겹쳐서...
방법 찾는 중...

designerbohyun replied:
아항~...어렵구만...

Lai-khan replied:
방법 찾았으~!
이쪽은 걱정 말어~

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+
+
+/* 입력을 다 안하고, 수정,등록버튼을 눌렀을 경우에 글씨 색상이 바뀌는  */
+.customer-name.before,
+.customer-number.before,
+.customer-address.before{
+    color: #eb0000;
+}
```
이거 스타일 적용 안됨
글씨 색깔 바꾸려고 한 것 같은데 안 바뀌어.

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3223 @@
+    color: #eb0000;
+}
+
+.name-input.before,
+.number-input.before,
+.address-input.before{
+    background-color: #ff000009;
+    border: 1px solid #FF0000;
+}
+
+.name-input.after,
+.number-input.after,
+.address-input.after{
+    background-color: #fff;
+    border: 1px solid #2657e9;
+}
```
이게 처음에는 `::before` `::after` 인줄 알았는데 아니더라고;;
내용이 입력 후에 after로 스타일을 바꾸는 거고,
focusout될때 내용이 없으면 before class를 붙여줘야 하는거고. 맞지?

이름이 직관적이지 않은것 같아. before, after면 하기전, 한 후 이렇게 생각하기 쉬운데 실제 동작을 보면 오히려 거꾸로니까. before, after 보다는 다른이름을 쓰는게 좋을 것 같아.
before -> 입력이 필요하다는 의미의 content-required 또는 input-required
after -> 입력이 되었다는 의미의 content-exist 또는 input-complete 어떨까?

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1261 @@
+                        <input class="address-input" type="text">
+                    </div>
+    
+                    <div class="customer-guide">
+                        <span class="required">*</span><p>는 필수 항목 입니다. (필수 항목을 입력해야 등록할 수 있습니다.)</p>
+                    </div>
```
혹시 입력값 검사하고 나서 오류 있으면 안내메시지 띄우는 방법 없을까?
alert 말고 하단에 빨간 글씨로.

여기도 좋구 아니면 input 태그 하단도 좋구

designerbohyun replied:
언니 내가 이 부분은 이해를 잘 못하는거 같은데, 혹시 안내메세지를 띄우는게 언니가 말해준 저 코드 구간처럼 보여지기를 원하는거야? 아니면 저 구간을 다른 방식으로 수정하는걸 원하는거야???? 그리고 입력값 검사하고 나서 오류있으면 안내 메시지를 띄운다는게 값을 입력하는 모든 구간에서 안내메시지가 뜨길 원하는 거야????

Lai-khan replied:
아, 원래는 잘못 입력하면 input 태그 아래에 조그맣게 빨간 글씨로 오류메시지가 나오기를 바랬는데
![image](https://user-images.githubusercontent.com/40736969/216971382-8e777015-92b4-48a8-b70b-866aa38fbdd7.png)

그냥 지금은 이렇게 해결했어! (electron 자체 errorDialog)
![image](https://user-images.githubusercontent.com/40736969/216971563-0c272616-c092-4eab-9c0d-31aa94818cef.png)

요부분은 패스해도 될 것 같아.

designerbohyun replied:
아~ 알겠어~!

Lai-khan reviewed (commented):
고객 정보 추가 개발중~

designerbohyun committed:
230207_github 이슈 + 클래스 수정

1. detail-icon 영역 이슈 : 너비 늘림
2. customer-address 이슈
: 50자에 맞춰서 늘어나다가 일정 높이가 되면 안 늘어남. + css로는 작업 무리라서 js로 추가 작업 진행
3. .before, .after 이슈
: before -> input-required
: after -> inpute-compelete로 수정
4. class 문제
: 동일한 스타일이 적용된 class 수정 및 일부 class 추가 작업 진행 -> 자세한 내용은 클래스 단위 설명란에 수정 작업 작성

Lai-khan commented on index.html:
```diff
@@ -779,7 +779,7 @@ <h3>고객 명단 추가</h3>
                     <div class="customer-address">
                         <span>주소</span>
-                        <input class="address-input" type="text">
+                        <textarea onkeydown="resize(this)" name="address-input" class="address-input autosize"></textarea>
```
![image](https://user-images.githubusercontent.com/40736969/217290058-f0f9ff06-a536-491a-a97f-83b2693cc070.png)
이게 딱 50자인데 뭔가 좀 어색하당..ㅎㅎ

그냥 저 주소 부분만 딱 input 사이즈를 늘려줄 수 있어? 50자에 맞게. (왼쪽두 같이!)
어차피 50자 이상은 입력하지 못하게 할테니까.
50자에 맞게 height를 늘려도 옆으로 가려나...

Lai-khan replied:
![image](https://user-images.githubusercontent.com/40736969/217290322-2bf0f826-1b66-49c0-971b-573ab7d8e4e9.png)
그리고 원래 이 스크롤바가 스크롤 할때만 보였던 것 같은데 갑자기 스타일 바꾸고 나서 항상 보이게 됐어.
수정 가능해?
스크롤바가 항시 있으니까 옆이 살짝 좁아보여서

Lai-khan replied:
아, 그리고 detail-icon은 범위 확실히 넓어졌어! 아주 좋더라! 👍

designerbohyun committed:
230213_customer-address 이슈 문제

1.  customer-address 이슈
: 높이 값은 수정해뒀는데, 차라리 고객 상세 정보 말고, 다른곳도 그냥 아예 높이값이 고정으로 보여지는게 통일성있게 보일 거 같아서 수정했어.

2. 스크롤바 문제는 좀 더 찾아봐야할 거 같긴한데.. 지금 내가 확인할 때는 그 부분을 건들이지 않아서.....지금 당장은 수정 방법이 없을 거 같아, 나중에 내가 찾으면 수정할게 그대신, 그냥 차라리 너비값을 조정했어. 지금 보니 너무 불필요하게 넓은 거 같아서! 차라리 조금 좁게 수정했어~!

Lai-khan commented on index.html:
```diff
@@ -779,7 +779,7 @@ <h3>고객 명단 추가</h3>
                     <div class="customer-address">
                         <span>주소</span>
-                        <textarea onkeydown="resize(this)" name="address-input" class="address-input autosize"></textarea>
+                        <textarea name="address-input" class="address-input autosize"></textarea>
```
![image](https://user-images.githubusercontent.com/40736969/218453510-a138a79c-421b-4548-9a1d-5a4f6feef853.png)

으음... 뭔가! 뭔가 1% 부족행...
혹시 가운데 정렬 가능? 높이 조금만 더 늘려주고

이게 1줄일때는 또 이렇게 치우쳐서  말야.
![image](https://user-images.githubusercontent.com/40736969/218453685-e03d0bfa-82ce-42a9-961c-2e067bfbbacb.png)

Lai-khan commented on css/style.css:
```diff
@@ -0,0 +1,3030 @@
+    cursor: pointer;
+}
+
+.delete-before-add::after{
+    background-color: #FF0000;
+}
```
이거 이렇게 하면 적용이 안되는디...ㅠ
`input-complete` 처럼 바꾸고 동의 체크박스 클릭할 때 class 값 추가하는 방식으로 갈께~
삭제 버튼에 delete-button id 값도 추가할게!
```suggestion
.delete-before-add.agree{
    background-color: #FF0000;
}
```

Lai-khan replied:
아, cursor 모양도 추가할게
```suggestion
.delete-before-add.agree {
    background-color: #FF0000;
    cursor: pointer;
}
```

위에 .agree 없는 건 `cursor: default;`로 바꿀게~

Lai-khan commented:
![image](https://user-images.githubusercontent.com/40736969/222912728-a8633483-b0ab-401d-8b9c-63c73c9a4aaa.png)
창이 최소 크기일때 이미지가 좀 커보이는데, 크기 줄어들면 이미지도 같이 줄어들 수 없을까?

![image](https://user-images.githubusercontent.com/40736969/222912869-3686be6a-f331-4073-a425-340e770e16de.png)
창이 좀 커지면 비율이 괜찮아지는데 작아지면 여백이 없어보여서..

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1243 @@
+
+
+                <!-- 고객 명단 클릭 안했을 경우, 어플 소개 페이지 -->
+                <div class="app-cover swiper mySwiper">
+                    <div class="app-cover-wrap swiper-wrapper">
+                      <div class="slide1 swiper-slide">
+                        <div class="img1">
+                            <img src="./img/slide1.png" alt="">
+                          </div>
+                      </div>
```
![image](https://user-images.githubusercontent.com/40736969/222912728-a8633483-b0ab-401d-8b9c-63c73c9a4aaa.png)
창이 최소 크기일때 이미지가 좀 커보이는데, 크기 줄어들면 이미지도 같이 줄어들 수 없을까?

![image](https://user-images.githubusercontent.com/40736969/222912869-3686be6a-f331-4073-a425-340e770e16de.png)
창이 좀 커지면 비율이 괜찮아지는데 작아지면 여백이 없어보여서..

Lai-khan replied:
그리고 저거 애니메이션? 이미지 3장 자동 슬라이드 되는거 안됨.
```
	display: -webkit-box;
	display: -ms-flexbox;
	display: -webkit-flex;
	display: flex;
```
개발자도구 보면 요기서 위에 3개 안먹히고 아래꺼 flex만 적용되고 있어. 한번 확인해줄래?

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1243 @@
+                <!-- 날짜 검색입력창 -->
+                <div class="date-search-bar">
+
+                    <div class="start-date-search-bar">
+                        <p>시작 날짜</p>
+                        <input type="date">
+                    </div>
+
+                    <div class="end-date-search-bar">
+                        <p>종료 날짜</p>
+                        <input type="date">
+                    </div>
```
![image](https://user-images.githubusercontent.com/40736969/226087529-4b0d8c31-c198-4372-98e5-e81ee6dc0eb9.png)

크기가 1200x800일때 깨진다.
1450까지 깨지는것 같애

Lai-khan replied:
![image](https://user-images.githubusercontent.com/40736969/226099558-775f092b-e964-4e90-ac3f-530544ff5f5c.png)
요거는 라이브러리 적용하면서 잘 수정했어!
신경 안써도 될 것 같아.

![image](https://user-images.githubusercontent.com/40736969/226099576-ebf24734-81b2-4e2e-92eb-f2e1802cc4d1.png)
요케 잘 뜬다.
달이랑 연도 클릭하면 수정 가능

designerbohyun committed:
230318_전반적인스타일오류문제

1. Customer-adrress 구간 : height 수정
2. search-bar: 사이즈 조정시 텍스트와 너비 오류 수정 작업
3. 서비스 소개 슬라이드 오류 : 너비 오류 + 애니메이션 작동 안함 -> 수정 완료

Lai-khan commented on css/style.css:
```diff
@@ -519,29 +519,27 @@
 
 /* 날짜 검색창 입력 */
 .date-search-bar{
```
날짜 검색창 부분은 내가 다른 라이브러리(primereact Calendar) 도입하면서 임의로 수정을 해서 이 css랑 좀 충돌한다.
요기 부터 날짜 검색창 부분은 반영 안할게.
파일은 그냥 수정된채로 놔둬줘. html 열어서 볼때 참고가능하니까.

Lai-khan commented on index.html:
```diff
@@ -1238,6 +1237,20 @@ <h3>관리항목 상세정보</h3>
 
     <!-- script 연결 : 고객 명단 어플 소개 페이지 - swiper -->
-    <script src="https://unpkg.com/swiper/swiper-bundle.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/swiper@9/swiper-bundle.min.js"></script>
+
+
+      <!-- Initialize Swiper -->
+    <script>
+        var swiper = new Swiper(".mySwiper", {
+        spaceBetween: 30,
+        centeredSlides: true,
+        autoplay: {
+            delay: 4000,
+            disableOnInteraction: false,
+        }
+        });
+    </script>
+
```
요건 react에 반영하려면 따로 Swiper 모듈을 설치해야하더라고. ([swiper  react](https://swiperjs.com/react))
요부분은 알아서 해볼게.

Lai-khan replied:
잘 적용했으!
근데 우리 예약 기능 없는데..?ㅎㅎ

Lai-khan commented on index.html:
```diff
@@ -915,17 +914,17 @@ <h3>고객 상세 정보</h3>
                     <div class="customer-address">
                         <span>주소</span>
-                        <textarea name="address-input" class="address-input autosize"></textarea>
+                        <textarea name="address-input" class="address-input autosize" placeholder="주소를 입력하세요"></textarea>
```
요기는 보여주기라 placeholder 없어도 되지 않아?ㅋㅋ

Lai-khan reviewed (commented):
일부만 알맞게 반영했으!

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1256 @@
+
+                
+                <!-- 삭제 버튼 클릭시 보이는 버튼 영역 -->
+                <div class="edit-button-wrap">
+                    <div class="edit-button-area">
+                        <button class="edit-button-area-cancel">
+                            <p>취소</p>
+                        </button>
+    
+                        <button class="edit-button-area-delete">
+                            <i style="margin-right:10px;" class="fa-solid fa-pen-to-square"></i>
+                            <p>수정</p>
+                        </button>
+                    </div>
+                </div>
```
![image](https://user-images.githubusercontent.com/40736969/228252870-320d564e-3b59-4e71-88ea-d78e15fdd809.png)
버튼 css가 이상해

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1256 @@
+            </div>
+
+                <!-- 삭제 버튼 클릭시 보이는 버튼 영역 -->
+                <div class="delete-button-wrap">
+                    <div class="delete-button-area">
+                        <button class="delete-button-area-cancel">
+                            <p>취소</p>
+                        </button>
+    
+                        <button class="delete-button-area-delete">
+                            <i  style="margin-right: 10px;" class="fa-solid fa-trash-can"></i>
+                            <p>삭제</p>
+                        </button>
+                    </div>
+                </div>
```
![image](https://user-images.githubusercontent.com/40736969/228253041-915a8ba9-e563-461a-a84f-337b216d7139.png)
여기도

Lai-khan reviewed (commented):
버튼 area css 문제
반응형 디자인 문제도 있는데 그건 직접 봐줘야 할것 같애

Lai-khan commented on index.html:
```diff
@@ -0,0 +1,1256 @@
+                        <!-- m-list -->
+                        <div class="manage-list">
+                            <li class="list list1">
+                                <div class="number-detail">
+                                    <p>10000</p> 
+                                </div>
+                                <!-- 삭제를 클릭하면, 숫자 대신 체크박스  -->
+                                <div class="number-checkbox-wrap">
+                                    <input class="checkbox" type="checkbox" name="checkbox">
+                                    <label for="checkbox"></label>
+                                </div>
+
+                                <!-- 수정을 클릭하면, 숫자 대신 라디오  -->
+                                <div class="number-radiobox-wrap">
+                                    <input class="radiobox" type="radio" name="radiobox">
+                                    <label for="radiobox"></label>
+                                </div>
```
여기 radio버튼이랑 체크박스 띄웠을때 UI 깨짐
![image](https://github.com/designerbohyun/crm/assets/40736969/0d5a0514-576e-45b2-9aba-d9897dc022ff)
![image](https://github.com/designerbohyun/crm/assets/40736969/518f8ef2-0411-4e20-ab90-5c3e7e804c30)
